src/app/_components/Card.tsx

```diff
@@ -21,38 +25,51 @@ type CardProps = {
   as?: React.ElementType
 }
 
-const borderStyles = {
-  'brand-300': 'border-brand-300',
-  'brand-400': 'border-brand-400',
-  'brand-500': 'border-brand-500',
-  'brand-600': 'border-brand-600',
-}
+type SpacingValue = keyof typeof theme.spacing
+type BreakpointValue = keyof typeof theme.screens
 
-function Link({
+type LeftProperty = `left-${SpacingValue}`
+type ResponsiveLeftProperty = `${BreakpointValue}:${LeftProperty}`
+
+type LinkProps = {
+  left?: LeftProperty | [LeftProperty, ResponsiveLeftProperty]
+} & CTAProps
+
+function renderTextIcon({
   href,
-  ariaLabel,
-  icon: Icon,
   text,
-  isExternalLink = false,
-}: CTAProps) {
+  icon: Icon,
+}: Pick<CTAProps, 'href' | 'text' | 'icon'>) {
+  const isExternal = isExternalLink(href)
+  const textElement = <span>{text}</span>
+
+  if (Icon) {
+    return [<Icon key="custom-icon" size={24} />, textElement]
+  }
+
+  if (isExternal) {
+    return [textElement, <ArrowUpRight key="arrow-icon" size={24} />]
```

Comment:
maybe we reverse the order on these 2 conditions? If, for example, we pass an book icon to external link, it will render the book icon and not the arrow icon. Not sure if we want the option to overwrite the arrow icon (for consistency sake)?

Reply:
Scratch that, we need to overwrite it. 😅 we have the google calendar case in governance